### PR TITLE
Chapter cards

### PR DESCRIPTION
**Pull request by AndreasGB**
Basically very similar to the Persons

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ChapterInfo.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.jellyfin.androidtv.model.itemtypes
+
+import org.jellyfin.apiclient.model.dto.ChapterInfoDto
+import org.jellyfin.apiclient.model.entities.ImageType
+
+class ChapterInfo(originalChapterInfoDto: ChapterInfoDto, val baseItem: BaseItem, index: Int) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ChapterInfo.kt]: Can't you get the image from the data inside ChapterInfoDto?

File: app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ImageCollection.kt [hunk 2]
```diff
@@ -15,14 +16,28 @@ class ImageCollection(original: BaseItemDto) {
 	val backdrops = original.backdropImageTags.map { Image(original.id, ImageType.Backdrop, it) }.toList()
 
 	class Image(private val itemId: String, private val type: ImageType, private val tag: String) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ImageCollection.kt]: Add the `index` option as nullable to the constructor

File: app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ImageCollection.kt [hunk 3]
```diff
@@ -15,14 +16,28 @@ class ImageCollection(original: BaseItemDto) {
...
+			if (type == ImageType.Chapter)
+				TvApp.getApplication().apiClient.GetImageUrl(itemId, ImageOptions().also {
+					it.imageType = type
+					it.tag = tag
+					it.imageIndex = index
+				})
+			else
 			TvApp.getApplication().apiClient.GetImageUrl(itemId, ImageOptions().also {
 				it.imageType = type
 				it.tag = tag
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ImageCollection.kt]: Just add the index here:
`if(imageIndex != null) it.index = imageIndex`

Might even be possible to do it without  the if statement (I think null is the default value)

File: app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ImageCollection.kt [hunk 4]
```diff
@@ -15,14 +16,28 @@ class ImageCollection(original: BaseItemDto) {
...
 			})
 		}
 
+		constructor(itemId: String, type: ImageType, tag: String, index: Int) : this(itemId, type, tag) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ImageCollection.kt]: Remove this constructor in favor of the other one I mentioned

File: app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ChapterInfo.kt (hunk 1 above)
Comment (AndreasGB) [reply app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ChapterInfo.kt]: Sadly not. A ChapterInfoDto doesn't contain the ID of the item it belongs to, and the index is also required by the server, inquiring with tag is not enough :/

Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ChapterInfo.kt]: Oh that's unfortunate. Can't do anything about it though :(

File: app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ImageCollection.kt [hunk 5]
```diff
@@ -14,15 +15,18 @@ class ImageCollection(original: BaseItemDto) {
...
 	val backdrops = original.backdropImageTags.map { Image(original.id, ImageType.Backdrop, it) }.toList()
 
-	class Image(private val itemId: String, private val type: ImageType, private val tag: String) {
+	class Image(private val itemId: String, private val type: ImageType, private val tag: String, private val index: Int? = null) {
+
 		val url: String by lazy {
 			TvApp.getApplication().apiClient.GetImageUrl(itemId, ImageOptions().also {
 				it.imageType = type
 				it.tag = tag
+				if (index != null) it.imageIndex = index
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ImageCollection.kt]: You can remove the if statement as the default value if `imageIndex` is null

File: app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ChapterInfo.kt (hunk 1 above)
Comment (AndreasGB) [reply app/src/main/java/org/jellyfin/androidtv/model/itemtypes/ChapterInfo.kt]: Yeh, bummed me out as well that this old API is sabotaging us here. At least it's clean on the lifted ones

**Pull request merged**